ledger/src/blockstore_processor.rs

```diff
@@ -1161,11 +1186,35 @@ fn load_frozen_forks(
                 leader_schedule_cache.set_root(new_root_bank);
                 new_root_bank.squash();
 
+                if let Some(snapshot_config) = snapshot_config {
+                    let block_height = new_root_bank.block_height();
+                    if block_height % snapshot_config.full_snapshot_archive_interval_slots == 0 {
+                        snapshot_utils::snapshot_bank(
+                            new_root_bank,
+                            new_root_bank.src.slot_deltas(&new_root_bank.src.roots()),
+                            &accounts_package_sender,
+                            &snapshot_config.bank_snapshots_dir,
+                            &snapshot_config.snapshot_archives_dir,
+                            snapshot_config.snapshot_version,
+                            snapshot_config.archive_format,
+                            None,
+                            Some(SnapshotType::FullSnapshot),
+                        )
+                        .expect("Failed to snapshot bank while loading frozen banks");
+                        trace!(
+                            "took bank snapshot for new root bank, block height: {}, slot: {}",
+                            block_height,
+                            *root
+                        );
+                        *last_full_snapshot_slot = Some(*root);
+                    }
+                }
```

Comment:
This is where snapshots are taken while processing blocks that are roots, and also sets the new value for `last_full_snapshot_slot`.

Reply:
TODO: call clean before snapshot_bank